API keys and access (#18)
vippsas/vipps-report-api: vipps-report-api-faq.md

draft documentation here to discuss with potential users of the API
and make sure we are developing the right solution.

## Which API keys give access to the API?

> In short: These API keys give access to the Report API:

Reviewer: These -> The following

Author: I disagree, but I'll giver you this one.

* The merchant's own API keys (the same API keys that are used to make payments)
* Partner keys (the same API keys that are used to make payments)
* Accounting partners' API keys (after the merchant has given the accounting partner access)

### The merchant's own API keys

The merchant's own API keys give full access to the Report API.

If the merchant uses an integration partner (see
[partner types](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/#partner-types)),
it is the same as using the merchant's own API keys.

When a merchant shares it's API keys for a sale unit with an integration partner,
Vipps has no way of knowing whether the API calls are made by the merchant or

> the integration partner.

Reviewer: When a merchant shares it's API keys for a sale unit with an integration partner,
Vipps has no way of knowing whether the API calls are made by the merchant or
the integration partner.*
---
How about rephrasing to:

If a merchant shares it's API keys for a sale unit with an integration partner,
Vipps has no way of knowing whether the API calls are made by the merchant or
the integration partner. This is why we mandate that all requests made by a partner on behalf of a Merchant is done by using the Partner's own keys.

Author: Good point. Added:
>This is why we require that all API requests made by a partner on behalf of a
merchant are done using the partner's own API keys (the partner keys).

### Partner keys

If the merchant has a sale unit that is configured with a platform partner (see
[partner types](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/#partner-types)),
the platform partner can use
[partner keys](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/partner-keys)

> to make requests to the Report API.

Reviewer 2: Could specify that this only grants access to the related ledger, but this might also be enough for now.

Author: Hmmm, yes, maybe refine later.

[Partner keys](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/partner-keys)
lets a partner make payments on behalf of a merchant, and the same API keys
also give access to the Report API.

## Specifying a accounting partner

In addition to the above, the merchant may add one or more accounting partners.
An accounting partner will get access to the Report API, but will not be allowed
to make payments, or move money in any way. The accounting partner can only see

> reports of the payments that have been made.

Reviewer 2: Pretty much same as above, "access to the Report API" might be enough for this part, but if the intended reader is a merchant it could be a bit too vague. But we can definitely just leave it like this for now.

Author: Ok, leaving as-is.

See:

> [Give access to an accounting partner](https://vippsas.github.io/vipps-developer-docs/docs/APIs/report-api/vipps-report-api#give-access-to-an-accounting-partner)

Reviewer 3: Can't this be a relative link since it's in the same repo?

Author: Yes! Fixed.

draft documentation here to discuss with potential users of the API
and make sure we are developing the right solution.

## Which API keys give access to the API?

In short: The following API keys give access to the Report API:
* The merchant's own API keys (the same API keys that are used to make payments)
* Partner keys (the same API keys that are used to make payments)

> * Accounting partners' API keys (after the merchant has given the accounting partner access)

Reviewer 4: I don't understand Accounting partners' API keys... are they:  (the API keys provided to the accounting partner after the merchant has given them access)?

> * Accounting partners' API keys (after the merchant has given the accounting partner access)

Reviewer 4: ```suggestion
* Accounting partners' API keys (the API keys provided to the accounting partner after the merchant has given the accounting partner access)
```

Reviewer 4: I don't understand Accounting partners' API keys... are they: (the API keys provided to the accounting partner after the merchant has given them access)?

Author: The accounting partner has its own API keys. The merchant gives the accounting partner access (on portal.vipps.no), and then the accounting partners keys are possible to use for the merchant's sale unit.

### The merchant's own API keys

The merchant's own API keys give full access to the Report API.

If the merchant uses an integration partner (see
[partner types](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/#partner-types)),
it is the same as using the merchant's own API keys.

When a merchant shares its API keys for a sale unit with an integration partner,
Vipps has no way of knowing whether the API calls are made by the merchant or
the integration partner.
This is why we require that all API requests made by a partner on behalf of a
merchant are done using the partner's own API keys (the partner keys).

### Partner keys

If the merchant has a sale unit that is configured with a platform partner (see
[partner types](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/#partner-types)),
the platform partner can use
[partner keys](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/partner-keys)

> to make requests to the Report API.

Reviewer 4: ```suggestion
to make requests to the Report API. This allows them to get information about payments and transactions associated with the ledger(s) they have access to, but does not give allow them to make payments or move money in any way.
```

Reviewer 2: but does not `give allow` them to make payments or move money in any way. One or the other?

Reviewer 4: oops! editing error

[Partner keys](https://vippsas.github.io/vipps-developer-docs/docs/vipps-partner/partner-keys)
lets a partner make payments on behalf of a merchant, and the same API keys
also give access to the Report API.

## Specifying a accounting partner

In addition to the above, the merchant may add one or more accounting partners.

> An accounting partner will get access to the Report API, but will not be allowed

Reviewer 4: ```suggestion
An accounting partner will get access to the Report API and be able to get information about payments and transactions associated with the ledger(s) they have access to. They will not be allowed
```
